src/components/duelling-picklist/picklist-item/picklist-item.component.tsx

```diff
@@ -10,15 +9,43 @@ import {
 import FocusContext from "../duelling-picklist.context";
 import Events from "../../../__internal__/utils/helpers/events";
 
-const PicklistItem = React.forwardRef(
+type Item = Record<string, unknown> | string | number;
+
+export interface PicklistItemProps {
+  /** Item content */
+  children: React.ReactNode;
+  /** Define if item is of type add or remove */
+  type: "add" | "remove";
+  /** Handler invoked when add/remove button is clicked or when space/enter is pressed on the whole item */
+  onChange: (item: Item) => void;
+  /** Value passed to the onChange handler */
+  item: Item;
```

Comment:
I noticed in storybook prop table it was a bit unclear what structure the `Item` type takes. What do you think about adding a sentence stating what values are accepted in the prop description?